src/util/dsp/levinson.cpp

```diff
@@ -0,0 +1,103 @@
+//**************************************************************************
+//
+//	$file: main.c $
+//	$author: Martin Fouilleul $
+//	$date: 28/11/2018 $
+//	$description: a simple Toeplitz system solver$
+//
+//**************************************************************************
+
+#include<string.h> // memset()
+
+namespace otto::dsp
+{
+//--------------------------------------------------------------------------
+//
+//	SolveToeplitz() is a (somewhat naive) implementation of the Levinson-Durbin algorithm for solving the Topelitz system Mx = y
+//
+//	arguments :
+//		- size : the size of the Toeplitz matrix and of vectors x and y
+//		- toeplitz : a buffer containing the first column of the matrix from bottom to top, then the first line from left to right.
+//			     eg. if the matrix is :
+//
+//			     [T0 T1   ...  Tn-1]
+//			     [T-1 T0  ...  Tn-2]
+//			     [...              ]
+//			     [T-n+1   ...    T0]
+//
+//			     you should pass the vector [T-n+1 T-n+2 ... T0 T1 ... Tn-1] of length (size+size-1)
+//
+//		- y : the vector at the right-hand side of the equation
+//		- x : result vector (that is, the x unknown at the left-hand side of the equation)
+//		- scratchBuffer : a block of memory used to hold intermediate computations. You must pass a buffer of size at least 4*size*sizeof(float).
+//
+//--------------------------------------------------------------------------
+
+void SolveToeplitz(unsigned int size, const float* toeplitz, const float* y, float* x, char* scratchBuffer)
```

Comment:
Maybe something like that would be more idiomatic to C++ ?

```suggestion
gsl::span<float> SolveToeplitz(const gsl::span<float> &toeplitz, const gsl::span<float> &y)
```

Reply:
Well, it's really as you prefer. 
It could be argued that the first version 'states' that there is one baseline size from which to deduce buffer lengths, although it doesn't enforce the lengths to be correct. 
On the other hand the second one doesn't enforce it either. You would still have to do a runtime  bound check to make it 'safe', so it really depends on the degree of trust you put in your user...